Configurations/unix-Makefile.tmpl

```diff
@@ -188,61 +188,111 @@ HTMLSUFFIX=html
+                                 (map { "-D".$_} @{$config{CPPDEFINES}}),
+                                 (map { "-I".$_} @{$config{CPPINCLUDES}}),
+                                 @{$config{CPPFLAGS}}) -}
+CFLAGS={- join(' ', @{$config{CFLAGS}}) -}
+CXXFLAGS={- join(' ', @{$config{CXXFLAGS}}) -}
+LDFLAGS= {- join(' ', @{$config{LDFLAGS}}) -}
+EX_LIBS= {- join(' ', @{$config{LDLIBS}}) -}
+
+MAKEDEPEND={- $config{makedepprog} -}
 
 PERL={- $config{perl} -}
 
-AR=$(CROSS_COMPILE){- $config{ar} -}
-ARFLAGS= {- join(' ', @{$config{arflags}}) -}
-RANLIB={- $config{ranlib} ? "\$(CROSS_COMPILE)$config{ranlib}" : "true"; -}
-RC= $(CROSS_COMPILE){- $target{rc} || "windres" -}
-RCFLAGS={- join(' ', @{$config{rcflags}}) -} {- $target{shared_rcflag} -}
+AR=$(CROSS_COMPILE){- $config{AR} -}
+ARFLAGS= {- join(' ', @{$config{ARFLAGS}}) -}
+RANLIB={- $config{RANLIB} ? "\$(CROSS_COMPILE)$config{RANLIB}" : "true"; -}
+RC= $(CROSS_COMPILE){- $config{RC} -}
+RCFLAGS={- join(' ', @{$config{RCFLAGS}}) -} {- $target{shared_rcflag} -}
+
+# We let the C compiler driver to take care of .s files. This is done in
+# order to be excused from maintaining a separate set of architecture
+# dependent assembler flags. E.g. if you throw -mcpu=ultrasparc at SPARC
+# gcc, then the driver will automatically translate it to -xarch=v8plus
+# and pass it down to assembler.
+AS={- $config{AS} ? "\$(CROSS_COMPILE)$config{AS}" : '$(CC) -c' -}
+ASFLAGS={- join(' ', @{$config{ASFLAGS}}) || '$(CFLAGS)' -}
+
 RM= rm -f
 RMDIR= rmdir
-TAR= {- $target{tar} || "tar" -}
-TARFLAGS= {- $target{tarflags} -}
-MAKEDEPEND={- $config{makedepprog} -}
+TAR= {- $target{TAR} || "tar" -}
+TARFLAGS= {- $target{TARFLAGS} -}
 
 BASENAME=       openssl
 NAME=           $(BASENAME)-$(VERSION)
 TARFILE=        ../$(NAME).tar
 
-# We let the C compiler driver to take care of .s files. This is done in
-# order to be excused from maintaining a separate set of architecture
-# dependent assembler flags. E.g. if you throw -mcpu=ultrasparc at SPARC
-# gcc, then the driver will automatically translate it to -xarch=v8plus
-# and pass it down to assembler.
-AS={- $config{as} ? "\$(CROSS_COMPILE)$config{as}" : '$(CC) -c' -}
-ASFLAGS={- join(' ', @{$config{asflags}}) || '$(CFLAGS)' -}
+##### Project flags ##################################################
+
+# Variables starting with CNF_ are common variables for all product types
+
+CNF_CPPFLAGS={- join(' ', $target{cppflags} || (),
+                          (map { "-D".$_} @{$config{defines}}),
+                          (map { "-I".$_} @{$config{includes}}),
+                          @{$config{cppflags}}) -}
+CNF_CFLAGS={- join(' ', $target{cflags} || (),
+                        @{$config{cflags}}) -}
+CNF_CXXFLAGS={- join(' ', $target{cxxflags} || (),
+                          @{$config{cxxflags}}) -}
+CNF_LDFLAGS={- join(' ', $target{lflags} || (),
+                         @{$config{lflags}}) -}
```

Comment:
Can you double-check if this is right? I'm a bit lost among them, but lflags is documented as "Flags that are used when linking apps.", but CNF_LDFLAGS is passed to LD_LDFLAGS, DSO_LDFLAGS and BIN_LDFLAGS.

Reply:
Thing is that `lflags` has been used *everywhere* where linking is involved, DSOs and shared libraries included.  If you look in 1.1.0, you'll see that it's passed in form of `$(LDFLAGS)` in all those cases.  So the docs aren't quite right.

For specialisation, there are other attributes prefixed with `lib_`, `shared_`, `dso_` and `bin_`...  or that's the idea at least.

Reply:
Maybe it would be appropriate to even rename it as ldflags, because it kind of stings the eye. Everything else is SOMETING = $target{lowercase(SOMETHING)}...

Reply:
I agree that it does.  But we're in the middle of a minor release, and I'm trying my best to keep a certain amount of backward compatibility, should anyone have made their own private config files (which we do encourage)...  so I think it's a no can do until 1.2.

(although, I *am* pondering some attribute name translations...  but I'll reserve that for another PR)